src/main/java/dk/cs/aau/huppaal/presentations/LogTabPresentation.java

```diff
@@ -0,0 +1,175 @@
+package dk.cs.aau.huppaal.presentations;
+
+import dk.cs.aau.huppaal.BuildConfig;
+import dk.cs.aau.huppaal.HUPPAAL;
+import dk.cs.aau.huppaal.controllers.LogTabController;
+import dk.cs.aau.huppaal.logging.Log;
+import dk.cs.aau.huppaal.logging.LogRegex;
+import dk.cs.aau.huppaal.logging.LogLinkQuantifier;
+import dk.cs.aau.huppaal.presentations.logging.Hyperlink;
+import dk.cs.aau.huppaal.presentations.logging.HyperlinkTextArea;
+import dk.cs.aau.huppaal.presentations.logging.TextStyle;
+import dk.cs.aau.huppaal.presentations.util.PresentationFxmlLoader;
+import dk.cs.aau.huppaal.utility.helpers.SelectHelper;
+import javafx.application.Platform;
+import javafx.beans.NamedArg;
+import javafx.beans.value.ChangeListener;
+import javafx.scene.control.Tooltip;
+import javafx.scene.layout.HBox;
+import org.fxmisc.flowless.VirtualizedScrollPane;
+
+import java.awt.Desktop;
+import java.io.File;
+import java.net.URI;
+import java.util.Optional;
+
+public class LogTabPresentation extends HBox {
+    public LogTabController controller;
+    public boolean autoscroll, wordwrap;
+    private final HyperlinkTextArea logArea;
+    private VirtualizedScrollPane<HyperlinkTextArea> scrollPane;
+    public LogTabPresentation(@NamedArg("textColor") String textColor) {
+        controller = PresentationFxmlLoader.loadSetRoot("LogTabPresentation.fxml", this);
+        wordwrap = true;
+        autoscroll = true;
+        logArea = new HyperlinkTextArea(this::onLinkClick);
+        scrollPane = new VirtualizedScrollPane<>(logArea);
+        initializeLogArea(textColor);
+        initializeButtons();
+        setupWordWrap();
+        setupAutoscroll();
+        Log.addOnLogAddedListener(this::onLogAdded);
+    }
+
+    private void onLinkClick(Hyperlink link) {
+        try {
+            var matcher = LogRegex.PATTERN.matcher(link.getLink());
+            if(!matcher.find()) // *should* never happen
+                throw new RuntimeException("Not a valid %s link pattern: '%s'".formatted(BuildConfig.NAME,link.getLink()));
+            // TODO: When syntactic elements have UUIDs to identify them, we should look at those IDs (instead/aswell)
```

Comment:
Does this refer to the UUIDs on Edges etc.?

Reply:
I have a plan to add a UUID to _all_ syntactic elements, just to avoid location id clashes (happens often when merging git branches). This todo is just a note to keep that in mind when introducing those UUIDs

For now, we'll live with being careful when merging and add the UUIDs in a separate PR 😄

Reply:
Sounds like a good addition, just wanted to make sure I understood what was being proposed :grin: